src/python/pants/option/global_options.py

```diff
@@ -1169,6 +1172,17 @@ class BootstrapOptions:
         help="Whether or not to use nailgun to run JVM requests that are marked as supporting nailgun.",
         advanced=True,
     )
+    interactive_process_graceful_shutdown_max_wait_time = IntOption(
+        "--interactive-process-graceful-shutdown-max-wait-time",
```

Comment:
This is the only place where we currently gracefully kill processes, but in the future we'll probably want to kill more things gracefully (test processes come to mind). Let's maybe not make this `interactive-process` specific... just `process`, namespaced with the other local process stuff.

While I could see having separate flags for interactive and non-interactive, this is/should-be the "big hammer" setting. Per-test or per-binary timeouts can come later and allow for more fine grained adjustment.